dev-utils/src/commands/run_command.rs

```diff
@@ -0,0 +1,91 @@
+use failure::{format_err, Error};
+use std::fs;
+use std::process::Command;
+use std::{thread, time};
+#[derive(Debug)]
+pub struct RunCommand {
+    chainid: String,
+    app_hash: String,
+}
+
+impl RunCommand {
+    pub fn new() -> Self {
+        RunCommand {
+            chainid: "".to_string(),
+            app_hash: "".to_string(),
+        }
+    }
+
+    fn get_tendermint_filename(&self) -> String {
+        format!(
+            "{}/.tendermint/config/genesis.json",
+            dirs::home_dir().unwrap().to_str().unwrap()
+        )
+        .to_string()
+    }
+    fn read_tendermint_genesis(&mut self) -> Result<(), Error> {
+        // check whether file exists
+        fs::read_to_string(&self.get_tendermint_filename())
+            .and_then(|contents| {
+                println!("current tendermint genesis={}", contents);
+                let json: serde_json::Value = serde_json::from_str(&contents).unwrap();
+                self.chainid = json["chain_id"].as_str().unwrap().to_string();
+                self.app_hash = json["app_hash"].as_str().unwrap().to_string();
+                println!("chainid={}", self.chainid);
+                println!("app_hash={}", self.app_hash);
+                Ok(())
+            })
+            .map_err(|_e| format_err!("read tendermint genesis error"))
+    }
+
+    pub fn run_program(&self, command: &str, arg: Vec<&str>) -> Result<(), Error> {
+        Command::new(command)
```

Comment:
How is it with the process stdout/stderr logs? I think it'll be useful to see what's going on, especially if it didn't work -- so one option is to dump/pipe the process logs into files, so one can inspect them later...
the other is not to use this custom launcher, but say https://github.com/DarthSim/overmind (this tool lets you specify concurrent processes in one "Procfile" and it'll run them similarly to Docker Compose)
@calvinaco any thoughts

Reply:
it's spawned process, so it's the same with running in multiple consoles.
so each daemon will store logs internally, it will work as the same.

Reply:
but "normally", the default behaviour of the particular spawned processes (tendermint, our abci app + tx validation server) is that they'd print to stdout... will that be lost?

Reply:
if it's necessary, we can pipe and copy , periodically collect, and remove oldest one.
i sugguest internal logging system instead of stdin/out.
because sometime , too many logs can burdern main system.(or precious main system disk space)

Reply:
this is a launcher for development purposes (if the developer's laptop is running out of disk space, I'm sure he can fix it :)).
what do you mean by "internal" logging? something like `journalctl` ? if so, then they'd need to be spawned e.g. as systemd services rather than plain processes

Reply:
For our own programs, the amount of logs can be controlled with `RUST_LOG` environment variable.
For tendermint, there's a "--log-level" flag:
```
      --log_level string   Log level (default "main:info,state:info,*:error")
```

Reply:
i'm thinking just writing to a file.
chain-abci.log
when it reaches some size, then backup 
chain-abci-2019-01-02.log
and start again

or i often use sqlite for logging

Reply:
working on with overmind

Reply:
dumping logs to files is fine as well in development -- just as long as there's a way for a developer to easily inspect what was going on in each process, it's fine

Reply:
ok

Reply:
overmind Procfile added,
now you can begin all stuff by using
overmind s